### PR TITLE
Superpmi mch merge

### PR DESCRIPTION
**Pull request by jashook**
Another requirement to use chem.

/cc @dotnet/jit-contrib.

### CODE REVIEW

File: scripts/superpmi.py [hunk 1]
```diff
@@ -554,6 +560,26 @@ def __merge_mc_files__(self):
         if not self.coreclr_args.skip_cleanup:
             for item in mc_contents:
                 os.remove(item)
+
+    def __merge_mch_files__(self):
+        """ Merge the mch files that were passed
+
+        Notes:
+            mcs -merge <s_baseMchFile> [self.coreclr_args.mch_files]
```
Comment (BruceForstall) [inline scripts/superpmi.py]: nit: you're using `-concat`, not `-merge` here

File: scripts/superpmi.py [hunk 2]
```diff
@@ -407,7 +410,10 @@ def collect(self):
                     self.__collect_mc_files__()
                 
                 if not self.coreclr_args.has_merged_mch:
-                    self.__merge_mc_files__()
+                    if not self.coreclr_args.merge_mch_files:
+                        self.__merge_mc_files__()
```
Comment (BruceForstall) [inline scripts/superpmi.py]: It seems to me like we should always merge whatever we need to, either mc or mch. I worry about adding this logic to distinguish between them.

Ideally, mcs `-merge` would subsume `-concat` and do any and all merging required (although the usage model would be a little more confusing if you also want to use `-recursive`).

**Comment by jkotas** [comment]
What is "chem"?

**Comment by BruceForstall** [comment]
@jkotas Internal tool by NET TI team

File: scripts/superpmi.py (hunk 2 above)
Comment (jashook) [reply scripts/superpmi.py]: I agree, I think this should be added to the backlog. Until then for expedience I will rely on the current functionality.

**Pull request merged**